core/roboconf-target-iaas-ec2/src/main/java/net/roboconf/target/ec2/internal/Ec2MachineConfigurator.java

```diff
@@ -124,16 +137,44 @@ public boolean configure() throws TargetException {
 
 		if( this.state == State.ASSOCIATE_ELASTIC_IP )
 			if( associateElasticIp())
-				this.state = State.ASSOCIATE_STORAGE;
-
-		if( this.state == State.ASSOCIATE_STORAGE )
-			if( associateStorage())
-				this.state = State.COMPLETE;
+				this.state = State.CREATE_VOLUME;
+		
+		if(volumeRequested()) {
```

Comment:
It would be better to wrap this section into its own function.
The **configure** method should only be a resume method. So, its structure must be as much simple as possible.